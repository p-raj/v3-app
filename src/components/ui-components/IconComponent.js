/**
 * Home Screen decides the layout of the components like
 * MembershipBar & RuntimeContainer.
 *
 * Lifting the state up, things have changed since
 * the last time this file created:
 * MembershipBar was supposed to hide when logged in a terminal.
 * No longer the case, we don't have any special case.
 * Every membership has Runtime associated with it.
 * Terminal - a type of membership now.
 *
 * Now, we need to show a Component showing,
 * no memberships assigned.
 * The current implementation will show this in 2 splits:
 * No memberships & no runtimes. :(
 *
 * */
import React  from 'react';
import RN, { Image, Text, TouchableHighlight, View } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import theme from '../common/theme'


const styles = RN.StyleSheet.create({
    iconContainer: {
        height: 95,
        width: 80,
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    iconShape: {
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center',

        // backgroundColor: theme.flatColorShades[Math.floor((Math.random() * 10) + 1)]

    },
    icon: {
        color: theme.iconColor,
        backgroundColor: "transparent",
        margin: 0,
        padding: 0
    },
    iconText: {
        color: theme.black,
        flex: 1,
        textAlign: 'center',
        fontSize: theme.h6,
        paddingTop: theme.paddingMinimal
    }
});


class IconComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {
        return (
            <View style={styles.iconContainer}>
                <TouchableHighlight
                    style={[styles.iconShape, {
                        width: this.props.iconSize || 60,
                        height: this.props.iconSize || 60,
                        backgroundColor: this.props.showImage ? 'white' : '#333'
                    }]}
                    onPress={this.props.onIconPress}>
                    {this.props.showImage ?
                        <Image source={{uri: this.props.imageUrl}}
                               width={75}
                               height={75}
                               style={{width: 75, height: 75, backgroundColor: 'white', alignSelf: 'center'}}
                               resizeMethod={'contain'}/> :
                        <Icon name={this.props.iconLogo} size={32}
                              style={styles.icon}/>}
                </TouchableHighlight>
                <Text numberOfLines={1} style={styles.iconText}>{this.props.iconText}</Text>
            </View>
        )
    };
}
IconComponent.propTypes = {
    iconText: React.PropTypes.string.isRequired,
    onIconPress: React.PropTypes.func.isRequired,
};


export default IconComponent;
