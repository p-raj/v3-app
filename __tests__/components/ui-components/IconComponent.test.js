import React from 'react';
import 'react-native';
import IconComponent from '../../../src/components/ui-components/IconComponent';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


test('IconComponent to display AppIcon and give onPress callback', () => {
    // Render an Icon with label
    const mockCallback = jest.fn();

    // The mock function is called twice
    const tree = renderer.create(
        <IconComponent iconText="App name" onIconPress={mockCallback}/>
    ).toJSON();
    expect(mockCallback.mock.calls.length).toBe(0);
    expect(tree).toMatchSnapshot();
});