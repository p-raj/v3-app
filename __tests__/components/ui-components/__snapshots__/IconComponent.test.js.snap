exports[`test IconComponent to display AppIcon and give onPress callback 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 80,
      "justifyContent": "space-between",
      "width": 80,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#333",
            "height": 60,
            "width": 60,
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 32,
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#FEFEFE",
            "margin": 0,
            "padding": 0,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      ?
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#191919",
        "flex": 1,
        "fontSize": 14,
        "paddingTop": 2,
        "textAlign": "center",
      }
    }>
    App name
  </Text>
</View>
`;
